backend/infrastructures/apps/extract/views/views.py

```diff
@@ -0,0 +1,109 @@
+from modules.load.services import commands as services_load_commands
+from modules.load.services import queries as load_queries
+from modules.transform.domain import commands as domain_transform_commands
+from modules.transform.services import commands as services_transform_commands
+
+logger = logging.getLogger(__name__)
+
+
+class ExtractViewSet(
+    ViewSet,
+):
+    @swagger_utils.extend_schema(
+        responses={
+            status.HTTP_201_CREATED: swagger_utils.OpenApiResponse(
+                description="Transformed and saved data successfully.",
+            ),
+            status.HTTP_400_BAD_REQUEST: swagger_utils.OpenApiResponse(
+                description="Issue occurred while processing dataset.",
+                response={
+                    "type": "object",
+                    "properties": {
+                        common_consts.ERROR_DETAIL_KEY: {
+                            "type": "string",
+                            "example": "",
+                        }
+                    },
+                },
+            ),
+        },
+    )
+    @inject.param(name="data_unit_of_work", cls="data_unit_of_work")
+    @inject.param(name="file_unit_of_work", cls="file_unit_of_work")
+    def create(
+        self,
+        request: Request,
+        data_unit_of_work: load_units_of_work.AbstractDataUnitOfWork,
+        file_unit_of_work: extract_units_of_work.AbstractFileUnitOfWork,
+    ) -> Response:
+        if "file" not in request.FILES:
+            return Response(
+                str({common_consts.ERROR_DETAIL_KEY: "File not attached."}),
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+        try:
+            logger.info("Extracting dataset...")
+            input_data: data_value_objects.InputData = service_extract_commands.extract(
+                domain_extract_commands.ExtractData(
+                    file_unit_of_work.file.save(
+                        bytes(request.FILES["file"].read()), request.FILES["file"].name
+                    )
+                )
+            )
+            logger.info("Dataset extracted.")
+
+            logger.info("Transforming dataset...")
+            output_data: list[
+                load_queries.OutputData
+            ] = services_transform_commands.transform(
+                domain_transform_commands.TransformData(input_data)
+            )
+            logger.info("Dataset transformed.")
+
+            logger.info("Saving dataset...")
+            services_load_commands.save(
+                data_unit_of_work, domain_load_commands.SaveData(output_data)
+            )
+        except FileNotFoundError as e:
+            logger.info(e.args[0])
+            return Response(
+                str({common_consts.ERROR_DETAIL_KEY: "File not found."}),
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+        except domain_exceptions.FileDataFormatNotSupportedException as e:
+            logger.info(str(e.args[0]))
```

Comment:
General rule.
When logging, add a sentence what when wrong, and then write why.

In this case:
logger.error("Can not extract data from file, due to not supported extension)